### PR TITLE
Optional Non-Entity Resource for handling inputs

### PR DESCRIPTION
**Pull request by huhlig**
This incorporates a monomorphizable resource that operates the same as the input component. This allows for input handling not attached to an entity. 

Example: 
```rust
#[derive(Actionlike, Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
pub enum DebugAction {
    ToggleConsole,
    ToggleOverlay,
    TogglePathing,
}

impl DebugAction {
    pub fn input_resource() -> InputResource<Self> {
        let mut res = InputResource::default();
        res.input_map
            .insert(DebugAction::ToggleConsole, KeyCode::F1);
        res.input_map
            .insert(DebugAction::ToggleOverlay, KeyCode::F2);
        res.input_map
            .insert(DebugAction::TogglePathing, KeyCode::F3);
        res
    }
}

#[derive(Debug, Default)]
pub struct DebugState {
    text_overlay: Option<TextOverlay>,
}

#[derive(Debug)]
struct TextOverlay {
    position: Entity,
    clock: Entity,
}

/// Should be called once to setup Gameplay Mode
pub fn handle_input(
    mut commands: Commands,
    mut debug_state: ResMut<DebugState>,
    mut asset_server: ResMut<AssetServer>,
    mut input_resource: Res<InputResource<DebugAction>>,
) {
    if input_resource
        .action_state
        .just_pressed(&DebugAction::ToggleOverlay)
    {
        if let Some(text_overlay) = &mut debug_state.text_overlay {
            trace!("Disable Debug Overlay");
            commands.entity(text_overlay.position).despawn();
            commands.entity(text_overlay.clock).despawn();
            debug_state.text_overlay = None;
        } else {
            trace!("Enable Debug Overlay");
            debug_state.text_overlay = Some(TextOverlay {
                position: commands
                    .spawn_bundle(TextBundle {
                        text: Text::with_section(
                            String::from("Position"),
                            TextStyle {
                                font: asset_server.load("fonts/spacemono.ttf"),
                                font_size: 30.0,
                                color: Color::WHITE,
                            },
                            TextAlignment {
                                vertical: VerticalAlign::Top,
                                horizontal: HorizontalAlign::Left,
                            },
                        ),
                        ..Default::default()
                    })
                    .id(),
                clock: commands
                    .spawn_bundle(TextBundle {
                        text: Text::with_section(
                            String::from("Clock"),
                            TextStyle {
                                font: asset_server.load("fonts/spacemono.ttf"),
                                font_size: 30.0,
                                color: Color::WHITE,
                            },
                            TextAlignment {
                                vertical: VerticalAlign::Bottom,
                                horizontal: HorizontalAlign::Right,
                            },
                        ),
                        ..Default::default()
                    })
                    .id(),
            });
        }
    }
}
```

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
As discussed on Discord, we should migrate https://github.com/Leafwing-Studios/leafwing_input_manager/blob/dev/examples/binding_menu.rs to use this pattern :)

**Comment by alice-i-cecile** [comment]
This will need to be revised to account for #106.

**Comment by Shatur** [comment]
Yep, just move your code after
```rust
    if disable_input.is_some() {
        return;
    }
```

File: src/lib.rs [hunk 1]
```diff
@@ -143,3 +143,20 @@ impl<A: Actionlike> Default for InputManagerBundle<A> {
         }
     }
 }
+
+/// Resource for Inputs
+pub struct InputResource<A: Actionlike> {
```
[the remark this answers is not in the document]
Comment (Shatur) [reply src/lib.rs]: I agree here. Things like `Res<InputResource<>>` look strange :)

**Comment by alice-i-cecile** [comment]
Superceded by #109.

**Pull request closed without merging**